snakemake_wrapper_utils/gatk.py

```diff
@@ -0,0 +1,99 @@
+import sys
+from snakemake_wrapper_utils.snakemake import is_arg
+
+
+def get_gatk_opts(
+    snakemake,
+    parse_arg_file=False,
+    parse_bam_index=False,
+    parse_bam_md5=False,
+    parse_vcf_index=False,
+    parse_vcf_md5=False,
+    parse_ref=False,
+    parse_ref_dict=False,
+    param_name="extra",
+):
+    """Obtain gatk_opts from input, output, params"""
+
+    gatk_opts = ""
+    extra = snakemake.params.get(param_name, "")
+
+    ##########################
+    ### Configuration file ###
+    ##########################
+
+    if parse_arg_file:
+        if is_arg("--arguments_file", extra):
+            sys.exit(
+                "You have specified an argument file (`--argument_file`) in `params.extra`; this is automatically inferred from `input.arg_file`."
+            )
+
+        # Multiple argument files can be provided. Order matters.
+        arg_file = snakemake.input.get("arg_file", "")
+        if arg_file:
+            if isinstance(arg_file, list):
+                arg_file = " --argument_file ".join(arg_file)
+
+            gatk_opts += f"--argument_file {arg_file}"
+
+    ######################
+    ### Reference file ###
+    ######################
+
+    if parse_ref:
+        if is_arg("-R", extra) or is_arg("--reference", extra):
+            sys.exit(
+                "You have specified reference file (`-R,--reference`) in `params.extra`; this is automatically inferred from `input.ref`."
+            )
+        ref = snakemake.input.get("ref")
+        if ref:
+            gatk_opts += f" --reference {ref} "
+
+    if parse_ref_dict:
+        if is_arg("--sequence-dictionary", extra):
+            sys.exit(
+                "You have specified reference sequence dictionary (`--sequence-dictionary`) in `params.extra`; this is automatically inferred from `input.ref_dict`."
+            )
+        ref_dict = snakemake.input.get("ref_dict", "")
+        if ref_dict:
+            gatk_opts += f" --sequence-dictionary {ref_dict} "
```

Comment:
Just for consistency with current wrappers.
```suggestion
                "You have specified reference sequence dictionary (`--sequence-dictionary`) in `params.extra`; this is automatically inferred from `input.dict`."
            )
        ref_dict = snakemake.input.get("dict", "")
        if ref_dict:
            gatk_opts += f" --sequence-dictionary {ref_dict}"
```